lib/stat_tracker.rb

```diff
@@ -14,34 +21,202 @@ def self.from_csv(data_files)
   end
 
   def game_collection
-    GameCollection.new
+    GameCollection.new(@games)
   end
 
   def team_collection
-    TeamCollection.new
+    TeamCollection.new(@teams)
   end
 
   def game_team_collection
-    GameTeamCollection.new
+    GameTeamCollection.new(@game_teams)
+  end
+
+  def game_collection_to_use
+    game_coll = instantiate_game_collection
+  end
+
+  def instantiate_game_collection
+    @instantiate_game_collection ||= game_collection.all
```

Comment:
Whatever you put at the right side of the pipes is what we're calling on/functionality. 
This is cool!